yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -321,7 +352,7 @@ export class Archiver implements ArchiveSource {
     );
 
     if (retrievedBlocks.length === 0) {
-      await this.store.setBlockSynchedL1BlockNumber(currentL1BlockNumber);
+      // We are not calling `setBlockSynchedL1BlockNumber` because it may cause sync issues if based off infura.
```

Comment:
Can you explain this comment?

Also, shouldn't we expect that we retrieve blocks if my local pending block num is less than what the rollup says?

Reply:
If you look higher up in the archiver there is a larger description around infura and consistency. So yes, while we SHOULD receive something. Infura might not return it to us because of the eventual consistency.